test/browser/call-joining.js

```diff
@@ -0,0 +1,129 @@
+        const res = await peerUser2.exec('getStartedCalls');
+        assert(res.some(c => c.callId === call.callId));
+    })
+
+    it('should add participant to call with addParticipantToCall and check they are a guest', async () => {
+        await Promise.all([
+            client.addParticipantToCall(call.callId, { userId: peerUser1.userId }),
+            expectEvents(client, [{
+                type: 'callStatus',
+                predicate: evt => evt.reason === 'participantAdded' && evt.call.callId === call.callId && evt.participant.userId === peerUser1.userId
+            }])
+        ]);
+        await sleep(3000);
+        await Promise.all([
+            peerUser1.exec('answerCall', call.callId, {audio: false, video: false}),
+            expectEvents(client, [{
+                type: 'callStatus',
+                predicate: evt => evt.reason === 'participantJoined' && evt.call.callId === call.callId && evt.participant.userId === peerUser1.userId && evt.participant.isMeetingGuest
+            }])
+        ]);
+    });
+
+    it('should allow other user to join the conference', async () => {
+        await Promise.all([
+            peerUser2.exec('joinConference', call.callId, {audio: true, video: false}),
+            expectEvents(client, [{
+                type: 'callStatus',
+                predicate: evt => evt.reason === 'participantJoined' && evt.call.callId === call.callId && evt.participant.userId === peerUser2.userId
+            }])
+        ]);
+    });
+
+    it('should get all calls and verify it contains the active call', async () => {
+        const res = await peerUser1.exec('getCalls');
+        assert(res && res.some(c => c.callId === call.callId));
+    });
+
+    it('should drop user from call and raise a callStatus event with reason: participantRemoved', async () => {
+        await Promise.all([
+            client.dropParticipant(call.callId, peerUser1.userId),
+            expectEvents(client, [{
+                type: 'callStatus',
+                predicate: evt => evt.reason === 'participantRemoved' && evt.call.callId === call.callId && !evt.call.participants.some(user => user.userId === peerUser1.userId)
+            }])
+        ]);
+    });
+
+    it('should allow user to leave conference and raise a callStatus event with reason: participantRemoved', async () => {
+        await Promise.all([
+            peerUser2.exec('leaveConference', call.callId),
+            expectEvents(client, [{
+                type: 'callStatus',
+                predicate: evt => evt.reason === 'participantRemoved' && evt.call.callId === call.callId && !evt.call.participants.some(user => user.userId === peerUser2.userId)
+            }])
+        ]);
+    });
+    
+    it('should add participant to call with addParticipant and check that they are NOT a guest', async () => {
+        await client.addParticipant(call.convId, [peerUser1.userId], true);
+        await sleep(3000);
```

Comment:
For any sleep we need, can you please add a comment why its needed?

Reply:
done.